aea/aea.py

```diff
@@ -269,7 +269,11 @@ def _get_msg_and_handlers_for_envelope(
         self, envelope: Envelope
     ) -> Tuple[Optional[Message], List[Handler]]:
         """Get the msg and its handlers."""
-        protocol = self.resources.get_protocol(envelope.protocol_id)
+        protocol = self.resources.get_protocol(envelope.protocol_specification_id)
```

Comment:
rename method to `get_protocol_by_specification_id` and maintain old method in parallel